test/helpers/deploy/deploy_resources.go

```diff
@@ -1010,6 +1010,80 @@ func KongDataPlaneClientCertificateAttachedToCP(
 	return cert
 }
 
+// KonnectExtension deploys a KonnectExtension.
+func KonnectExtension(
+	t *testing.T,
+	ctx context.Context,
+	cl client.Client,
+	opts ...ObjOption,
+) *konnectv1alpha1.KonnectExtension {
+	t.Helper()
+
+	ke := &konnectv1alpha1.KonnectExtension{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "konnect-extension-",
+		},
+	}
+	for _, opt := range opts {
+		opt(ke)
+	}
+
+	require.NoError(t, cl.Create(ctx, ke))
+	logObjectCreate(t, ke)
+	return ke
+}
+
+// ObjectSupportingKonnectConfiguration defines the interface of types supporting setting `KonnectConfiguration`.
+type ObjectSupportingKonnectConfiguration interface {
+	*konnectv1alpha1.KonnectGatewayControlPlane |
+		*konnectv1alpha1.KonnectExtension |
+		*konnectv1alpha1.KonnectCloudGatewayNetwork
+}
+
+// WithKonnectConfiguration returns an option function that sets the `KonnectConfiguration` in the object.
+func WithKonnectConfiguration[T ObjectSupportingKonnectConfiguration](
+	konnectConfiguration konnectv1alpha1.KonnectConfiguration,
+) ObjOption {
+	return func(obj client.Object) {
+		switch o := any(obj).(type) {
+		case *konnectv1alpha1.KonnectGatewayControlPlane:
+			o.Spec.KonnectConfiguration = konnectConfiguration
+		case *konnectv1alpha1.KonnectExtension:
+			o.Spec.KonnectConfiguration = lo.ToPtr(konnectConfiguration)
+		case *konnectv1alpha1.KonnectCloudGatewayNetwork:
+			o.Spec.KonnectConfiguration = konnectConfiguration
+		}
+	}
+}
+
+// KonnectExtensionWithAPIAuthRefAndCPID deploys a KonnectExtension attaching to control plane
+// by the CP's ID and the API auth configuration to the Konnect server where the CP is in.
+func KonnectExtensionWithAPIAuthRefAndCPID(
+	t *testing.T,
+	ctx context.Context,
+	cl client.Client,
+	apiAuth konnectv1alpha1.KonnectAPIAuthConfigurationRef,
+	cpID string,
+	opts ...ObjOption,
+) *konnectv1alpha1.KonnectExtension {
```

Comment:
With these other functions and options in place I believe we want to remove this one, no? (especially since it appears unused now)

Reply:
Done.